essentials-framework/Essentials Core/PepperDashEssentialsBase/Routing/RoutingInterfaces.cs

```diff
@@ -111,10 +111,70 @@ public interface IRmcRouting : IRoutingNumeric
         IntFeedback AudioVideoSourceNumericFeedback { get; }
     }
 
-	/// <summary>
+
+    /// <summary>
+    /// Defines an IRmcRouting with a feedback event 
+    /// </summary>
+    public interface ITxRoutingWithFeedback : ITxRouting, IRoutingNumericFeedback
+    {
+    }
+
+    /// <summary>
+    /// Defines an IRmcRouting with a feedback event 
+    /// </summary>
+    public interface IRmcRoutingWithFeedback : IRmcRouting, IRoutingNumericFeedback
+    {
+    }
+
+    /// <summary>
 	/// Defines an IRoutingOutputs devices as being a source - the start of the chain
 	/// </summary>
 	public interface IRoutingSource : IRoutingOutputs
 	{
 	}
+
+    /// <summary>
+    /// Defines an event structure for reporting output route data
+    /// </summary>
+    public interface IRoutingNumericFeedback : IKeyName
+    {
+        event EventHandler NumericSwitchChange;

```

Comment:
Close! But this will call an event with the delegate that looks like `void EventHandler(object publisher, EventArgs args)`. You want it to look like `void EventHandler(object publisher, RoutingNumericEventArgs args)` so that the subscriber doesn't have to cast to the correct class to access the information. If you change the event to be `event EventHandler<RoutingNumericEventArgs> NumericSwitchChange`, that'll take care of that for you.

Reply:
Rectified - along with some suggestions from @ndorin about making this more in line with the rest of Essentials.

Now the event reports both ints for ins/outs, but also RoutingPorts for ins/outs.

It also makes use of the RoutingPort FeedbackMatchObject.